src/fields/field_algo_base.impl.hh

```diff
@@ -53,26 +53,31 @@ string FieldAlgorithmBase<spacedim, Value>::template_name() {
 
 
 template <int spacedim, class Value>
-it::AbstractRecord FieldAlgorithmBase<spacedim, Value>::input_type
-    = it::AbstractRecord("Field:"+template_name(), "Abstract record for all time-space functions.")
-          .allow_auto_conversion("FieldConstant");
-
-
-
-template <int spacedim, class Value>
-Input::Type::AbstractRecord FieldAlgorithmBase<spacedim, Value>::get_input_type(const typename Value::ElementInputType *element_input_type) {
-    it::AbstractRecord type= it::AbstractRecord("Field:"+template_name(), "Abstract record for all time-space functions.");
-    type.allow_auto_conversion("FieldConstant");
-
-    FieldConstant<spacedim,Value>::get_input_type(type, element_input_type);
-    FieldFormula<spacedim,Value>::get_input_type(type, element_input_type);
+Input::Type::AbstractRecord & FieldAlgorithmBase<spacedim, Value>::get_input_type(const typename Value::ElementInputType *element_input_type) {
+	const it::Selection *element_input_sel = nullptr;
+	if (element_input_type != nullptr) {
+		const it::Selection * sel_type = dynamic_cast<const it::Selection *>(element_input_type);
+		if (sel_type != NULL ) element_input_sel = sel_type;
+	}
+
+	it::AbstractRecord type= it::AbstractRecord("Field:"+template_name(), "Abstract record for all time-space functions.")
+    	.allow_auto_conversion("FieldConstant")
+		.set_element_input(element_input_sel)
+		.close();
+
+	if ( !type.is_finished() ) {
```

Comment:
It's necessary. Only 'is_closed' test adds childs repeated for each call and throws exception.
This solution is temporary.